### PR TITLE
feat(scan): consider .gitignore to automatically exclude paths by default

### PR DESCRIPTION
**Pull request by rafaela-soares**
Issue #4584 will be closed when the branch `release/1.6` is merged into the master.

**Proposed Changes**
- consider `.gitignore` to automatically exclude paths by default

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

**Comment by roi-orca** [comment]
@rafaela-soares I would suggest having the default behavior as of today, meaning .gitignore content will be taken into consideration as part of the scan, and the newly added flag, which can be called `"--include-gitignore"` will exclude the content from the scan.

If we do that, we do not break the current usage of the CLI, and we are having a "strict by default" approach

File: docs/commands.md [hunk 1]
```diff
@@ -106,6 +107,9 @@ Global Flags:
 
 The other commands have no further options.
 
+## Exclude Paths
+By default, KICS now automatically excludes paths according to the .gitignore file. If you want to disable this behavior, you use the flag `--add-gitignore`.
```
Comment (nunoocx) [inline docs/commands.md:111]: Change to
By default, KICS excludes paths specified in the .gitignore file in the root of the repository. To disable this behavior, use flag `--add-gitignore`.

File: docs/dockerhub.md [hunk 2]
```diff
@@ -53,6 +53,7 @@ Usage:
   kics scan [flags]
 
 Flags:
+      --add-gitignore                 disables the usage of .gitignore file to automatically exclude paths
```
Comment (nunoocx) [inline docs/dockerhub.md:56]: disables the exclusion of paths specified within .gitignore file

File: e2e/fixtures/assets/scan_help [hunk 3]
```diff
@@ -2,6 +2,7 @@ Usage:
   kics scan [flags]
 
 Flags:
+      --add-gitignore                 disables the usage of .gitignore file to automatically exclude paths
```
Comment (nunoocx) [inline e2e/fixtures/assets/scan_help:5]: same as above

File: internal/console/assets/scan-flags.json [hunk 4]
```diff
@@ -185,5 +185,11 @@
     "defaultValue": "",
     "usage": "case insensitive list of platform types to scan\n(${supportedPlatforms})",
     "validation": "validateMultiStrEnum"
+  },
+  "add-gitignore": {
+    "flagType": "bool",
+    "shorthandFlag": "",
+    "defaultValue": "false",
+    "usage": "disables the usage of .gitignore file to automatically exclude paths"
```
Comment (nunoocx) [inline internal/console/assets/scan-flags.json:193]: same as above

**Review by nunoocx — CHANGES_REQUESTED** [review]
Please review the suggestions

**Comment by rafaela-soares** [comment]
> @rafaela-soares I would suggest having the default behavior as of today, meaning .gitignore content will be taken into consideration as part of the scan, and the newly added flag, which can be called `"--include-gitignore"` will exclude the content from the scan.
> 
> If we do that, we do not break the current usage of the CLI, and we are having a "strict by default" approach

Hello, @roi-orca!

Thank you so much for being involved in this question and for your suggestion.

It was a product decision to exclude paths specified within .gitignore file by default. A few users suggested that since the .gitignore file presents the files that will be not committed. Since it can be considered a breaking change, this new behavior will be only included from version 1.6. To disable this behavior, you can use flag `--add-gitignore`.

File: docs/commands.md [hunk 5]
```diff
@@ -43,6 +43,7 @@ Usage:
   kics scan [flags]
 
 Flags:
+      --add-gitignore                 disables the exclusion of paths specified within .gitignore file
```
Comment (nunoocx) [inline docs/commands.md:46]: Hi @rafaela-soares , revisiting this, the flag add-gitignore sounds misleading.
Taking the suggestion from @roi-orca (and considering that the default will be to include gitignore) WDYT if we change to `--exclude-gitignore` ?
If agreed, please make the remainder of the code consistent to that new naming.

**Review by nunoocx — CHANGES_REQUESTED** [review]

**Review by nunoocx — APPROVED** [review]
Great work!

**Pull request merged**